src/components/grouped-bar-plot/GroupedBarPlot.js

```diff
@@ -1,52 +1,65 @@
 /* eslint-disable react/prop-types */
 /* eslint-disable react/jsx-filename-extension */
 import React from 'react';
-import { BarChart } from 'recharts';
-import Bar from '../shared/bar/Bar';
+import { BarChart, CartesianGrid, Legend, Tooltip, XAxis, YAxis } from 'recharts';
 import {
   BAR_STROKE_WIDTH,
+  groupedBarDisplayTypes,
   PLOT_COLORS,
   PLOT_SECONDARY_COLORS,
 } from '../../constants/plotConstants';
 import useTooltip from '../../hooks/useTooltip';
+import Bar from '../shared/bar/Bar';
 
 import {
+  getCategoryAxisDataKey,
   getCategoryValueAxes,
+  getCategoryValueAxisByDataKey,
   getData,
+  getGroupedBarPlotDisplayType,
   getIsDataPivoted,
-  getIsSeriesStacked,
   getMargin,
+  getUniqueValuesOfDataKey,
+  getXAxis,
+  getXAxisDataKey,
+  getYAxis,
   getYAxisDataKey,
+  getYAxisTickFormatter,
+  pivotDataByDataKey,
 } from '../../utils/plotConfigGetters';
 import GeneralChartComponents from '../general-chart-components/GeneralChartComponents';
 import PlotContainer from '../plot-container/PlotContainer';
+import { overrideAxisConfig } from '../../utils/overrideAxisConfig';
 
 function PivotedGroupedBar({
   plotConfig = {},
   updateHoveredItemId = () => {},
   updateClickedItemId = () => {},
   hoveredItemId = null,
 }) {
-  const data = getData(plotConfig);
-  const yAxisDataKey = getYAxisDataKey(plotConfig);
-  return data.map((series, index) => {
-    return Bar({
-      id: series.name,
-      data: series.data,
-      stroke: PLOT_COLORS[index % PLOT_COLORS.length],
+  const displayType = getGroupedBarPlotDisplayType(plotConfig);
+  const isSeriesStacked = displayType === groupedBarDisplayTypes.STACKED;
+  const categoryAxisDataKey = getCategoryAxisDataKey(plotConfig);
+  const uniqueValuesOfCategoryKey = getUniqueValuesOfDataKey(plotConfig, categoryAxisDataKey);
+
+  return uniqueValuesOfCategoryKey.map((value, index) =>
```

Comment:
`value` is equal to `series.name`, which came before, they were both the unique values of the category axis.